### PR TITLE
Perf manifest parsing

### PR DESCRIPTION
**Pull request by carneyweb**
Improve normal performance by only initializing the deprecated properties if the deprecated apis are used by doing a deep copy of m_Headers into m_PropertiesDeprecated. Otherwise m_PropertiesDeprecated is never created. These changes will be deleted completely when the deprecated apis are completely removed.

### CODE REVIEW

File: framework/src/bundle/BundleManifest.cpp [hunk 1]
```diff
@@ -143,34 +177,41 @@ bool BundleManifest::Contains(const std::string& key) const
...
+  if (m_Headers.cend() != iter)
+  {
     return iter->second;
   }
   return Any();
 }
 
 Any BundleManifest::GetValueDeprecated(const std::string& key) const
 {
+  copy_deprecated_properties(m_Headers, m_PropertiesDeprecated);
```
Comment (jeffdiclemente) [inline framework/src/bundle/BundleManifest.cpp]: there is a possible race condition if more than one thread calls a deprecated bundle property API. I'd suggest using ```std::call_once``` for all the calls to ```copy_deprecated_properties```

File: framework/src/bundle/BundleManifest.cpp [hunk 2]
```diff
@@ -143,34 +177,41 @@ bool BundleManifest::Contains(const std::string& key) const
...
+  if (m_PropertiesDeprecated.cend() != iter)
+  {
     return iter->second;
   }
   return Any();
 }
 
 std::vector<std::string> BundleManifest::GetKeysDeprecated() const
 {
+  copy_deprecated_properties(m_Headers, m_PropertiesDeprecated);
```
Comment (jeffdiclemente) [inline framework/src/bundle/BundleManifest.cpp]: there is a possible race condition if more than one thread calls a deprecated bundle property API. I'd suggest using ```std::call_once``` for all the calls to ```copy_deprecated_properties```

File: framework/src/bundle/BundleManifest.cpp [hunk 3]
```diff
@@ -143,34 +177,41 @@ bool BundleManifest::Contains(const std::string& key) const
...
+       ++iter)
+  {
     keys.push_back(iter->first);
   }
   return keys;
 }
 
 std::map<std::string, Any> BundleManifest::GetPropertiesDeprecated() const
 {
+  copy_deprecated_properties(m_Headers, m_PropertiesDeprecated);
```
Comment (jeffdiclemente) [inline framework/src/bundle/BundleManifest.cpp]: there is a possible race condition if more than one thread calls a deprecated bundle property API. I'd suggest using ```std::call_once``` for all the calls to ```copy_deprecated_properties```

File: framework/src/bundle/BundleManifest.cpp [hunk 4]
```diff
@@ -143,34 +177,41 @@ bool BundleManifest::Contains(const std::string& key) const
...
   std::vector<std::string> keys;
-  for (AnyMap::const_iterator iter = m_PropertiesDeprecated.begin();
-       iter != m_PropertiesDeprecated.end();
-       ++iter) {
+  for (AnyMap::const_iterator iter = m_PropertiesDeprecated.cbegin();
+       iter != m_PropertiesDeprecated.cend();
+       ++iter)
+  {
```
Comment (jeffdiclemente) [inline framework/src/bundle/BundleManifest.cpp]: The formatting of the if statement does not follow convention.

File: framework/src/bundle/BundleManifest.cpp [hunk 5]
```diff
@@ -143,34 +177,41 @@ bool BundleManifest::Contains(const std::string& key) const
 Any BundleManifest::GetValue(const std::string& key) const
 {
   auto iter = m_Headers.find(key);
-  if (iter != m_Headers.end()) {
+  if (m_Headers.cend() != iter)
+  {
```
Comment (jeffdiclemente) [inline framework/src/bundle/BundleManifest.cpp]: The formatting of the if statement does not follow convention.

File: cmake/CMakeResourceDependencies.cpp [hunk 6]
```diff
@@ -20,14 +20,12 @@
```
Comment (jeffdiclemente) [inline cmake/CMakeResourceDependencies.cpp]: was this file accidentally brought in to the PR? This change doesn't appear to be relevant to manifest parsing.

File: framework/src/bundle/BundleManifest.cpp (hunk 1 above)
Comment (carneyweb) [reply framework/src/bundle/BundleManifest.cpp]: Done.

File: framework/src/bundle/BundleManifest.cpp (hunk 3 above)
Comment (carneyweb) [reply framework/src/bundle/BundleManifest.cpp]: Done.

File: framework/src/bundle/BundleManifest.cpp (hunk 5 above)
Comment (carneyweb) [reply framework/src/bundle/BundleManifest.cpp]: Done.

File: framework/src/bundle/BundleManifest.cpp (hunk 4 above)
Comment (carneyweb) [reply framework/src/bundle/BundleManifest.cpp]: Done.

File: framework/src/bundle/BundleManifest.cpp (hunk 2 above)
Comment (carneyweb) [reply framework/src/bundle/BundleManifest.cpp]: Done.

File: cmake/CMakeResourceDependencies.cpp (hunk 6 above)
Comment (carneyweb) [reply cmake/CMakeResourceDependencies.cpp]: I believe this came in as part of a merge of 349_Fix_Makefile_builds_on_macos.

Comment (jeffdiclemente) [reply cmake/CMakeResourceDependencies.cpp]: this file should be removed from the PR then.

File: framework/test/gtest/BundleManifestTest.cpp [hunk 7]
```diff
@@ -36,6 +36,59 @@
```
Comment (jeffdiclemente) [inline framework/test/gtest/BundleManifestTest.cpp]: this file should be formatted using the clang.format file in the root source directory

File: cmake/CMakeResourceDependencies.cpp (hunk 6 above)
Comment (carneyweb) [reply cmake/CMakeResourceDependencies.cpp]: Hopefully it's now removed. I restored the state of the file to be as it was before the merge.
